cityofaustin/atd-prefect
# Remove protection of certain columns in the CRIS import script

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/10652. 

NB: this PR is into `moped-project` which is a feature branch we're currently running out of for our atd-data03 production deployment. I will rebase the PR to merge into main if needed.

Testing:

* In your VZ local stack, checkout the `layer-data-model` branch if needed
* Spin up a local DB and populate it with data
* Create the `cris` schema, as found here: [create_cris_schema.sql](https://github.com/cityofaustin/atd-vz-data/blob/import-cris-data/atd-toolbox/materials_from_ldm_conversion/create_cris_schema.sql)
* Back in this repo, spin up a container of the VZ dockerized-agent / execution environment and attach to it
* Use the CRIS extract account to cause CRIS to push an export of data for some time interval of your choosing to the SFTP endpoint
* When the file is available, either copy it to your local machine and adjust the tasks used in the flow to pick it up directly (at or around [line 485 in the cris_import.py script](https://github.com/cityofaustin/atd-prefect/blob/import-cris-data-ldm/flows/vision-zero/cris_import/cris_import.py#L484-L489)), or let the flow take care of pulling it down from the SFTP site on its own. It can be useful to keep a copy somewhere on the SFTP endpoint so you can put it back to rerun the flow a second time if you need to.
* Run the flow and observe that it will now create updates when processing the CRIS extract and does so with no consideration as to what is an "important" or editable field in the VZ. 
  * There should be no way for an update from CRIS to cause the VZ conflict resolution system to be used.
